src/Cli.Tests/ConfigureOptionsTests.cs

```diff
@@ -150,5 +150,195 @@ public void TestUpdateDepthLimitForGraphQL(int? newDepthLimit)
+        [DataRow("cosmosdb_postgresql", DisplayName = "Update the database type to CosmosDB_PGSQL")]
+        [DataRow("mysql", DisplayName = "Update the database type to MySQL")]
+        public void TestDatabaseTypeUpdate(string dbType)
+        {
+            // Arrange
+            _fileSystem!.AddFile(TEST_RUNTIME_CONFIG_FILE, new MockFileData(INITIAL_CONFIG));
+
+            Assert.IsTrue(_fileSystem!.File.Exists(TEST_RUNTIME_CONFIG_FILE));
+
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config));
+            Assert.IsNotNull(config.Runtime);
+
+            // Act
+            ConfigureOptions options = new(
+                dataSourceDatabaseType: dbType,
+                config: TEST_RUNTIME_CONFIG_FILE
+            );
+            Assert.IsTrue(TryConfigureSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert
+            string updatedConfig = _fileSystem!.File.ReadAllText(TEST_RUNTIME_CONFIG_FILE);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config.Runtime);
+            Assert.AreEqual(config.DataSource.DatabaseType, Enum.Parse<DatabaseType>(dbType, ignoreCase: true));
+        }
+
+        /// <summary>
+        /// Tests the update of the database type from CosmosDB_NoSQL to MSSQL in the runtime config.
+        /// This method verifies that the database type can be changed from CosmosDB_NoSQL to MSSQL and that the 
+        /// specific MSSQL option 'set-session-context' is correctly added to the configuration and the specific
+        /// cosmosDB options are removed. 
+        /// </summary>
+        [TestMethod]
+        public void TestDatabaseTypeUpdateCosmosDB_NoSQLToMSSQL()
+        {
+            // Arrange
+            _fileSystem!.AddFile(TEST_RUNTIME_CONFIG_FILE, new MockFileData(INITIAL_COSMOSDB_NOSQL_CONFIG));
+
+            Assert.IsTrue(_fileSystem!.File.Exists(TEST_RUNTIME_CONFIG_FILE));
+
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(INITIAL_COSMOSDB_NOSQL_CONFIG, out RuntimeConfig? config));
+            Assert.IsNotNull(config.Runtime);
+
+            // Act
+            ConfigureOptions options = new(
+                dataSourceDatabaseType: "mssql",
+                dataSourceOptionsSetSessionContext: true,
+                config: TEST_RUNTIME_CONFIG_FILE
+            );
+            Assert.IsTrue(TryConfigureSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert
+            string updatedConfig = _fileSystem!.File.ReadAllText(TEST_RUNTIME_CONFIG_FILE);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config.Runtime);
+            Assert.AreEqual(config.DataSource.DatabaseType, DatabaseType.MSSQL);
+            Assert.AreEqual(config.DataSource.Options!.GetValueOrDefault("set-session-context", false), true);
```

Comment:
what does the `false` value do here: `Options!.GetValueOrDefault("set-session-context", false), true);`

Reply:
it's the default value, it will give false if the key is not found.